compiler/src/iree/compiler/Codegen/LLVMGPU/test/config_matvec.mlir

```diff
@@ -0,0 +1,52 @@
+
+#pipeline_layout = #hal.pipeline.layout<push_constants = 0, sets = [
+  #hal.descriptor_set.layout<0, bindings = [
+    #hal.descriptor_set.binding<0, storage_buffer>,
+    #hal.descriptor_set.binding<1, storage_buffer>,
+    #hal.descriptor_set.binding<2, storage_buffer>
+  ]>
+]>
+
+hal.executable @dynamic_batch_matvec {
+hal.executable.variant @rocm target(<"rocm", "rocm-hsaco-fb", {target_arch = "gfx1100"}>) {
+    hal.executable.export @dynamic_batch_matvec layout(#pipeline_layout)
+    builtin.module {
+      func.func @dynamic_batch_matvec() {
+        %c32_i64 = arith.constant 32 : i64
+        %cst = arith.constant 0.000000e+00 : f16
+        %0 = hal.interface.constant.load[0] : i32
+        %1 = hal.interface.constant.load[1] : i32
+        %2 = hal.interface.constant.load[2] : i32
+        %3 = hal.interface.constant.load[3] : i32
+        %4 = hal.interface.constant.load[4] : i32
+        %13 = arith.index_castui %0 : i32 to index
+        %18 = arith.index_castui %1 : i32 to index
+        %19 = arith.index_castui %2 : i32 to index
+        %24 = arith.index_castui %3 : i32 to index
+        %29 = arith.index_castui %4 : i32 to index
+        %30 = hal.interface.binding.subspan set(0) binding(2) type(storage_buffer) alignment(64) offset(%19) : !flow.dispatch.tensor<writeonly:tensor<32x1x128xf16>>
+        %31 = flow.dispatch.workload.ordinal %24, 0 : index
+        %32 = flow.dispatch.workload.ordinal %29, 1 : index
+        %33 = hal.interface.binding.subspan set(0) binding(0) type(storage_buffer) alignment(64) offset(%13) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<32x1x?xf16>>{%31}
+        %34 = hal.interface.binding.subspan set(0) binding(1) type(storage_buffer) alignment(64) offset(%18) flags(ReadOnly) : !flow.dispatch.tensor<readonly:tensor<32x?x128xf16>>{%32}
+        %35 = flow.dispatch.tensor.load %33, offsets = [0, 0, 0], sizes = [32, 1, %31], strides = [1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<32x1x?xf16>>{%31} -> tensor<32x1x?xf16>
+        %36 = flow.dispatch.tensor.load %34, offsets = [0, 0, 0], sizes = [32, %32, 128], strides = [1, 1, 1] : !flow.dispatch.tensor<readonly:tensor<32x?x128xf16>>{%32} -> tensor<32x?x128xf16>
+        %37 = tensor.empty() : tensor<32x1x128xf16>
+        %38 = linalg.fill ins(%cst : f16) outs(%37 : tensor<32x1x128xf16>) -> tensor<32x1x128xf16>
+        %39 = linalg.batch_matmul ins(%35, %36 : tensor<32x1x?xf16>, tensor<32x?x128xf16>) outs(%38 : tensor<32x1x128xf16>) -> tensor<32x1x128xf16>
+        flow.dispatch.tensor.store %39, %30, offsets = [0, 0, 0], sizes = [32, 1, 128], strides = [1, 1, 1] : tensor<32x1x128xf16> -> !flow.dispatch.tensor<writeonly:tensor<32x1x128xf16>>
+        return
+      }
+    }
+  }
+}
+
+//   CHECK-DAG: #[[$CONFIG:.+]] = #iree_codegen.lowering_config<tile_sizes = {{\[}}[1, 1, 1], [0, 0, 0, 32]{{\]}}>
```

Comment:
This should be 64. But it should be fixed after fixing the subgroup size for ROCm in general I think. So fine for me now. @kuhar

Reply:
Are we using 32 even for gfx90a/gfx940 and such? Those are natively subgroup size 64 right? 

For gfx1100 I think it's a bit fuzzy given that we can set either 32 or 64..

Reply:
I was under the impression it was natively 32. If that's not the case, then, yeah we should definitely be picking 64.

Reply:
ok you're right, it should be 64.

Reply:
Ok, double reason to find the patch I put together before that enabled subgroup size control in rocm.

Reply:
Though for GCN the native SIMD size is 16.. Is the recommendation to use 32 in general for ROCm?

Reply:
Yeah. We need to ask what it happens given the underlying SIMD has 16 lanes, and the native subgroup size is 64. I'm just unclear what it happens when we are forcing using 32 here. Does it effectively disable half of the compute lanes? Or is it actually only launch 2 SIMDs so it's fine?

Reply:
GCN/CDNA native is 64, RDNA native is 32 with 64 allowed via double issue.

Reply:
I will land a configuration change for this separately